components/camera/replaypcd/replaypcd_utils_test.go

```diff
@@ -41,6 +43,7 @@ const (
 // can be overwritten to allow developers to trigger desired behaviors during testing.
 type mockDataServiceServer struct {
 	datapb.UnimplementedDataServiceServer
+	httpMock []*httptest.Server
```

Comment:
I agree with Jeremy that a more accurate mock of the server would be if there was just one handler that provided different files based on the request. Since this is just a test I'm okay with leaving it as-is, but I do think changing it could make the test setup easier to understand (I personally was a bit confused by how it worked at first). Adding a comment to the mock server explain the request and response could also help

Reply:
kk I can add some comments to try making it clear what the setup is doing!